hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/functional/FutureIO.java

```diff
@@ -114,6 +117,70 @@ public static <T> T awaitFuture(final Future<T> future,
     }
   }
 
+  /**
+   * Given a future, evaluate it.
+   * <p>
+   * Any exception generated in the future is
+   * extracted and rethrown.
+   * </p>
+   * @param collection collection of futures to be evaluated
+   * @param <T> type of the result.
+   * @return the list of future's result, if all went well.
+   * @throws InterruptedIOException future was interrupted
+   * @throws IOException if something went wrong
+   * @throws RuntimeException any nested RTE thrown
+   */
+  public static <T> List<T> awaitFuture(final Collection<Future<T>> collection)
```

Comment:
wondering where will this be used without timeout?

Reply:
There is a method `public static <T> T awaitFuture(final Future<T> future)` in the class for single future. Added this method in order to keep new methods in sync with existing methods.